reframe/core/modules.py

```diff
@@ -939,3 +941,111 @@ def emit_load_instr(self, module):
 
     def emit_unload_instr(self, module):
         return ''
+
+
+class SpackImpl(ModulesSystemImpl):
+    '''Base class for TMod Module system (Tcl).'''
+
+    def __init__(self):
+        # Try to figure out if we are indeed using the TCL version
+        try:
+            completed = osext.run_command('spack -V')
+        except OSError as e:
+            raise ConfigError(
+                'could not find a sane Spack installation') from e
+
+        self._version = completed.stdout.strip()
+
+        # self._version = version
+        # try:
+        #     # Try the Python bindings now
+        #     completed = osext.run_command(self.modulecmd())
+        # except OSError as e:
+        #     raise ConfigError(
+        #         'could not get the Python bindings for TMod: ' % e) from e
+
+        # if re.search(r'Unknown shell type', completed.stderr):
+        #     raise ConfigError(
+        #         'Python is not supported by this TMod installation')
+
+    def name(self):
+        return 'spack'
+
+    def version(self):
+        return self._version
+
+    def modulecmd(self, *args):
+        return ' '.join(['spack', *args])
+
+    def _execute(self, cmd, *args):
+        modulecmd = self.modulecmd(cmd, *args)
+        completed = osext.run_command(modulecmd)
+        if re.search(r'Error', completed.stderr) is not None:
+            raise SpawnedProcessError(modulecmd,
+                                      completed.stdout,
+                                      completed.stderr,
+                                      completed.returncode)
```

Comment:
You should simply check the error code. Tmod32 is just misbehaving not returning a proper exit code when it fails, that's why we do this hack. In your case, you should simply execute `run_command()` with `check=True`.